addon/patches/store.js

```diff
@@ -43,6 +43,16 @@ Store.reopen({
   },
 
   didSaveRecord(internalModel, { data }) {
+    if (data === undefined) {
+      data = {
+        meta: undefined,
+        _top_level_meta: undefined
+      }
+    } else if (!('meta' in data)) {
+      data.meta = undefined;
+      data._top_level_meta = undefined;
+    }
+    
     internalModel.recordReference.__update_meta(data.meta);
     internalModel.recordReference.__update_responseMeta(data._top_level_meta);
```

Comment:
I think the current implementation might be a little too complex and hard to read. What do you think of guarding if `data` is defined like:

```js
internalModel.recordReference.__update_meta(data && data.meta);
internalModel.recordReference.__update_responseMeta(data && data._top_level_meta);
```